### ISSUE TITLE
denyUpdate and denyInsert as functions

### ISSUE THREAD

**Issue by andreivolt** [aldeed/meteor-collection2#205]
`denyUpdate` and `denyInsert` are very useful for fine-grained restrictions on records, but since they are not functions, it's not possible to conditionally grant permission to some users, etc.

It seems pretty straightforward to implement, is there any reason why it's not ?

**Comment by aldeed** [comment]
I'm not sure how I feel about putting security in the schema in general. I think it makes more sense to do the property-level security along with other security on the server, like with [ongoworks:security](https://github.com/ongoworks/meteor-security).

I would look at a PR that adds this, but I'm not going to prioritize it.

**Comment by andreivolt** [comment]
Thanks for your reply!

Yeah, I was hesitant about this too, that's why I asked of your opinion.

I had looked at ongoworks:security and I my initial impression was that it seemed a bit too complicated for simple rules like "only the owner of this document should be able to update this property". I'm sure there are plenty of ways to shoot oneself in the foot, in regard to complexity with this approach, but it would so damn convenient for simple situations like the one I described above.

For now I resorted using Meteor.methods for such operations.
